cpp/ql/src/Security/CWE/CWE-311/CleartextTransmission.ql

```diff
@@ -18,6 +18,21 @@ import semmle.code.cpp.valuenumbering.GlobalValueNumbering
 import semmle.code.cpp.models.interfaces.FlowSource
 import DataFlow::PathGraph
 
+/**
+ * A DataFlow node corresponding to a variable or function call that
+ * might contain or return a password or other sensitive information.
+ */
+class SensitiveNode extends DataFlow::Node {
+  SensitiveNode() {
+    this.asExpr() = any(SensitiveVariable sv).getInitializer().getExpr() or
+    this.asExpr().(VariableAccess).getTarget() =
+      any(SensitiveVariable sv).(GlobalOrNamespaceVariable) or
+    this.asUninitialized() instanceof SensitiveVariable or
+    this.asParameter() instanceof SensitiveVariable or
+    this.asExpr().(FunctionCall).getTarget() instanceof SensitiveFunction
+  }
```

Comment:
Could we use `SsaDefinition.getAnUltimateDefiningValue` to clean up some of this?

Reply:
Does SSA have any modelling of uninitialized variables?

Reply:
Hm, good point. I don't think so, no. Maybe it won't cut down the number of cases that much after all.